win_build/CUDA 11.8.props

```diff
@@ -0,0 +1,228 @@
+    </PropertyGroup>
+
+    <PropertyGroup>
+        <!-- Check CUDA Toolkit versioned environment variable. -->
+        <CudaToolkitDir Condition="'$(CudaToolkitDir)' == ''">$(CudaToolkitVersionedPath)</CudaToolkitDir>
+
+        <!-- Last, check the new CUDA Toolkit style/Nsight toolkit registry value. -->
+        <CudaToolkitDir Condition="'$(CudaToolkitDir)' == ''">$([MSBuild]::GetRegistryValueFromView('HKEY_LOCAL_MACHINE\SOFTWARE\NVIDIA Corporation\GPU Computing Toolkit\Cuda\$(CudaToolkitVersion)', 'InstallDir', null, RegistryView.Registry64, RegistryView.Registry32))</CudaToolkitDir>
+
+        <!-- Ensure $(CudaToolkitDir) has a trailing backslash. -->
+        <CudaToolkitDir Condition="'$(CudaToolkitDir)' != '' AND !HasTrailingSlash('$(CudaToolkitDir)')">$(CudaToolkitDir)\</CudaToolkitDir>
+
+        <CudaToolkitIncludeDir Condition="'$(CudaToolkitIncludeDir)' == ''">$(CudaToolkitDir)include</CudaToolkitIncludeDir>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(Platform)' == 'Win32' or '$(Platform)' == 'x86'">
+        <CudaToolkitBinDir Condition="'$(CudaToolkitBinDir)' == ''">$(CudaToolkitDir)bin</CudaToolkitBinDir>
+
+        <CudaToolkitLibDir Condition="'$(CudaToolkitLibDir)' == '' AND Exists('$(CudaToolkitDir)lib\Win32')">$(CudaToolkitDir)lib\Win32</CudaToolkitLibDir>
+        <CudaToolkitLibDir Condition="'$(CudaToolkitLibDir)' == ''">$(CudaToolkitDir)lib</CudaToolkitLibDir>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(Platform)' == 'x64'">
+        <CudaToolkitBinDir Condition="'$(CudaToolkitBinDir)' == '' AND Exists('$(CudaToolkitDir)bin64')">$(CudaToolkitDir)bin64</CudaToolkitBinDir>
+        <CudaToolkitBinDir Condition="'$(CudaToolkitBinDir)' == ''">$(CudaToolkitDir)bin</CudaToolkitBinDir>
+
+        <CudaToolkitLibDir Condition="'$(CudaToolkitLibDir)' == '' AND Exists('$(CudaToolkitDir)lib\x64')">$(CudaToolkitDir)lib\x64</CudaToolkitLibDir>
+        <CudaToolkitLibDir Condition="'$(CudaToolkitLibDir)' == ''">$(CudaToolkitDir)lib64</CudaToolkitLibDir>
+    </PropertyGroup>
+
+    <PropertyGroup>
+        <CudaToolkitNvccPath Condition="'$(CudaToolkitNvccPath)' == ''">$(CudaToolkitBinDir)\nvcc.exe</CudaToolkitNvccPath>
+
+        <!--
+            2010 is no longer supported, continue to map the command-line arg
+            and let nvcc generate the error.
+          -->
+        <CudaClVersion Condition="'$(PlatformToolset)' == 'v100'">2010</CudaClVersion>
+        <CudaClVersion Condition="'$(PlatformToolset)' == 'Windows7.1SDK'">2010</CudaClVersion>
+        <CudaClVersion Condition="'$(PlatformToolset)' == 'v110'">2012</CudaClVersion>
+        <CudaClVersion Condition="'$(PlatformToolset)' == 'v120'">2013</CudaClVersion>
+        <CudaClVersion Condition="'$(PlatformToolset)' == 'v140'">2015</CudaClVersion>
+        <CudaClVersion Condition="'$(PlatformToolset)' == 'v141'">2017</CudaClVersion>
+        <CudaClVersion Condition="'$(PlatformToolset)' == 'v142'">2019</CudaClVersion>
```

Comment:
I believe here should be `CudaClVersion` set for v143 (2022).
We don't support building on older versions, so you can remove others.
Also, we don't support x86 builds anymore, so you can remove this as well

Reply:
Currently this is more or less the unmodified .props file from the CUDA toolkit and I hesitated to edit the XML before I break the build of the CUDA kernel or the property editor in the project properties.
Shall I remove it? If so, I will try to do this in the next step.

Reply:
Ideally, keep the current file that is on master and apply minimal required changes.

Reply:
> Ideally, keep the current file that is on master and apply minimal required changes.

There were a lot of changes in those files (e.g. the .targets file is 10x bigger than the currently used file). Additional options, targets and also new project properties (CUDA C/C++ and CUDA Linker) were added. There is also a CUDA installation path that is used for the CUDA installation now. It is not as simple as just reuse the old version of it.
For the ease of use I would prefer to keep those files as original as possible so it is easier to upgrade to newer versions of the CUDA toolkit.

Reply:
I don't agree with that. We have a minimalistic file that works. And it has no additional parameters that are not used. And it's much easier to support (and update) and understand file that has no unused stuff.
Just copypaste file and hope it works is not a good way